File: Quick Starts/Logging/VB/LoggingQuickStart/EventInformationForm.vb
'===============================================================================
' Microsoft patterns & practices Enterprise Library
' Logging Application Block QuickStart
'===============================================================================
' Copyright © Microsoft Corporation.  All rights reserved.
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
' OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
' LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
' FITNESS FOR A PARTICULAR PURPOSE.
'===============================================================================

Public Class EventInformationForm

    Private _eventID As Integer = -1
    Private _categories As ICollection(Of String) = New List(Of String)(0)

    '''<summary>
    ''' Event ID for event to be logged.
    ''' </summary>
    Public ReadOnly Property EventId() As Integer
        Get
            Return EventId
        End Get
    End Property
   

    ''' <summary>
    ''' Message for event to be logged.
    ''' </summary>
    Public ReadOnly Property Message() As String
        Get
            Return messageTextbox.Text
        End Get
    End Property
 
    ''' <summary>
    ''' Priority of event to be logged.
    ''' </summary>
    Public ReadOnly Property Priority() As Integer
        Get
            Return Decimal.ToInt32(priorityNumericUpDown.Value)
        End Get
    End Property


    ''' <summary>
    ''' Collection of categories for the event.
    ''' </summary>
    Public ReadOnly Property Categories() As ICollection(Of String)
        Get
            Return _categories
        End Get
    End Property
        

    Private Sub EventInformationForm_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        messageTextbox.Focus()
    End Sub

    Private Sub CancelEntryButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles CancelEntryButton.Click
        Me.DialogResult = Windows.Forms.DialogResult.Cancel
        Me.Close()
    End Sub

    Private Sub OkButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles OkButton.Click
        RecordSelectedCategories()

        If ValidateInput() Then
            _eventID = Convert.ToInt32(eventIdTextBox.Text)
            Me.DialogResult = Windows.Forms.DialogResult.OK
            Me.Close()
        End If
    End Sub

    Private Sub RecordSelectedCategories()
        If traceCheckBox.Checked Then Categories.Add("Trace")
        If debugCheckbox.Checked Then Categories.Add("Debug")
        If generalCheckbox.Checked Then Categories.Add("General")
        If uiCheckbox.Checked Then Categories.Add("UI Events")
        If dataAccessCheckbox.Checked Then Categories.Add("Data Access Events")
        If troubleshootingCheckbox.Checked Then Categories.Add("Troubleshooting")
    End Sub

    ''' <summary>
    ''' Validate contents for the event ID text box, the priority text box and the category.
    ''' </summary>
    ''' <returns></returns>
    Private Function ValidateInput() As Boolean
        Dim errorMessage As String = My.Resources.InvalidDataMessage
        Dim validationError As Boolean = False

        Try
            Dim id As Integer = Convert.ToInt32(eventIdTextBox.Text)
        Catch
            errorMessage += My.Resources.InvalidEventIDMessage
            validationError = True
        End Try
        If (validationError) Then

            MessageBox.Show(errorMessage, My.Resources.QuickStartTitleMessage, MessageBoxButtons.OK, MessageBoxIcon.Error)
            Return False
        End If

        Return True
    End Function

End Class

File: Quick Starts/Logging/CS/LoggingQuickStart/EventInformationForm.cs
//===============================================================================
// Microsoft patterns & practices Enterprise Library
// Logging Application Block QuickStart
//===============================================================================
// Copyright © Microsoft Corporation.  All rights reserved.
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
// OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
// LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
// FITNESS FOR A PARTICULAR PURPOSE.
//===============================================================================

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace LoggingQuickStart
{
    public partial class EventInformationForm : Form
    {
        private ICollection<string> categories = new List<string>(0);
        private int eventID = -1;

        public EventInformationForm()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Event ID for event to be logged.
        /// </summary>
        public int EventId
        {
            get { return this.eventID; }
        }

        /// <summary>
        /// Message for event to be logged.
        /// </summary>
        public string Message
        {
            get { return this.messageTextbox.Text; }
        }

        /// <summary>
        /// Priority of event to be logged.
        /// </summary>
        public int Priority
        {
            get { return Decimal.ToInt32(this.priorityNumericUpDown.Value); }
        }

        /// <summary>
        /// Collection of categories for the event.
        /// </summary>
        public ICollection<string> Categories
        {
            get { return this.categories; }
        }

        private void EventInformationForm_Load(object sender, EventArgs e)
        {
            this.messageTextbox.Focus();
        }

        private void CancelButton_Click(object sender, EventArgs e)
        {
            base.DialogResult = DialogResult.Cancel;
            base.Close();
        }

        private void OkButton_Click(object sender, EventArgs e)
        {
            RecordSelectedCategories();

            if (ValidateInput())
            {
              this.eventID = Convert.ToInt32(this.eventIdTextBox.Text);
              base.DialogResult = DialogResult.OK;
              base.Close();
            }
        }

        private void RecordSelectedCategories()
        {
            if (this.traceCheckBox.Checked)
            {
                categories.Add("Trace");
            }
            if (this.debugCheckbox.Checked)
            {
                categories.Add("Debug");
            }
            if (this.generalCheckbox.Checked)
            {
                categories.Add("General");
            }
            if (this.uiCheckbox.Checked)
            {
                categories.Add("UI Events");
            }
            if (this.dataAccessCheckbox.Checked)
            {
                categories.Add("Data Access Events");
            }
            if (this.troubleshootingCheckbox.Checked)
            {
                categories.Add("Troubleshooting");
            }
        }

        /// <summary>
        /// Validate contents for the event ID text box, the priority text box and the category.
        /// </summary>
        /// <returns></returns>
        private bool ValidateInput()
        {
          string errorMessage = Properties.Resources.InvalidDataMessage;
          bool validationError = false;

          try
          {
            int id = Convert.ToInt32(this.eventIdTextBox.Text);
          }
          catch (Exception)
          {
            errorMessage += Properties.Resources.InvalidEventIDMessage;
            validationError = true;
          }

          if (validationError)
          {
            MessageBox.Show(errorMessage, Properties.Resources.QuickStartTitleMessage, MessageBoxButtons.OK, MessageBoxIcon.Error);
            return false;
          }

          return true;
        }


    }
}